src/stdlib/mod.rs

```diff
@@ -163,3 +169,127 @@ fn assert<B: Backend>(
 
     Ok(None)
 }
+
+fn to_bits<B: Backend>(
+    compiler: &mut CircuitWriter<B>,
+    generics: &GenericParameters,
+    vars: &[VarInfo<B::Field, B::Var>],
+    span: Span,
+) -> Result<Option<Var<B::Field, B::Var>>> {
+    // should be two input vars
+    assert_eq!(vars.len(), 2);
+
+    // but the better practice would be to retrieve the value from the generics
+    let num_var = generics.get("LEN");
+
+    // num should be greater than 0
+    assert!(num_var > 0);
+
+    // alternatively, it can be retrieved from the first var, but it is not recommended
+    // let num_var = &vars[0];
+
+    let var_info = &vars[1];
+    let var = &var_info.var;
+    assert_eq!(var.len(), 1);
+
+    // second var is the value to convert
+    let val = match &var[0] {
+        ConstOrCell::Cell(cvar) => cvar,
+        ConstOrCell::Const(_) => todo!(),
+    };
+
+    // convert value to bits
+    let mut bits = Vec::new();
+    let mut e2 = 1u32;
```

Comment:
`Field::one()` otherwise it will overflow

Reply:
(as `u32` is 32-bit, but we could have values up til ~250 bits in some of these curves)